src/test/java/com/gargoylesoftware/htmlunit/WebRequestTest.java

```diff
@@ -425,4 +428,19 @@ public void getRequestParametersFromUrlEncodedBodyPost() throws Exception {
 
         assertEquals(0, request.getRequestParameters().size());
     }
+
+    @Test
```

Comment:
i like to have a bit more detailed test - checking the values of the new normalized version
and maybe also testing the normalization - having a parameter with null here or two separate tests

Reply:
I elaborated a bit on the tests. Please have a look at the new commit. If you think i should add some more testing please let me know.

Reply:
fine for me